### ISSUE TITLE
install into a project

### ISSUE THREAD

**Issue by DragonSai** [master-technology/AssetManagerStudio#31]
So this asset manager is totally amazing.

Thank you very much for building it.

I clicked the asset, Install into a project and select the project, then click Download... both buttons turn grey.

I click another asset, Install into a project and select the project, then click Download... both buttons turn grey.

Wait patiently as they download and do their thing

Open up the project and only the first asset is there.

I select the asset downloaded but not installed then select install into a project , select the project and nothing...

The button beside it says delete not download.

Nothing turns grey.

I originally thought is was because I had 4.27.2 installed. The asset manager was having issues finding UE5.3.2.

I renamed ue4.27 folder so the asset manager wouldn't find it, making UE5.3.2 the new default engine.

I will delete the second asset and do what I did with the first to see if that works.

Just a note. The download and install UE5.3.2 downloaded and failed to install... I use Slackware, I am not surprised it failed.

I went to github and got UE5.3.2 and did what I need to get it running manually.

The asset manager has no problems recognizing githubs version.

I know something isn't quite right with multiple downloads cause I installed most of Brushify and Brushify SmartBrushes. 

Everything but SmartBrushes installed. I deleted, reinstalled/ downloaded smartbrushes and it installed.

Oh and for the record, Plugins have to be built in with the engine manually, although the asset manager did recognise them

download and attempted to install. I didn't try install to project... opposed to install to engine, which I did. I thought you 

had a fancy way of installing plugins with linux... I had to try cause this is really an awesome creation.

All that being said... I love the hints the best. I learned a lot of little things I never knew.

**Comment by NathanaelA** [comment]
Thanks for the feedback....
 
>> I clicked the asset, Install into a project and select the project, then click Download... both buttons turn grey.
>> I click another asset, Install into a project and select the project, then click Download... both buttons turn grey.

That is correct, once you start a download of an asset the buttons go grey to say they are in the download queue.   I do think I need to have a screen/pulldown showing the download queue, because I think this can confuse people on what is set to download.  It only downloads one asset at a time, once the first one is done, then it will start on the second, etc.
- Having a Queue window is on my TODO list...  ;-)

You should be able to find any assets that has downloaded or failed to download in the vault.  If it failed to download, if you start the asset download again it will RESUME from where it failed.   
-  Auto-Resuming is also on my TODO list.

---

>> The button beside it says delete not download.

That is correct, because everything in the "Vault" is  downloaded, however it is possible that it wasn't downloaded fully, I will run some tests on partially downloaded assets to see if they show up as full assets...  

---

>> I select the asset downloaded but not installed then select install into a project , select the project and nothing...

It should have installed into the project, unless the asset wasn't fully downloaded...  Hmm, that could be why, I should mark assets in a partial downloaded state with a X over them or something letting you know that it hasn't completed downloading...

----

>> I originally thought is was because I had 4.27.2 installed. The asset manager was having issues finding UE5.3.2.

It should find all engines as long as they are in a common directory.    In the settings you can set the parent directory of all your engines (or even multiple common directories delimited with a `|`) and it will scan for all engines and should see them all and display them on the "Engines" tab.        On the Engines tab you can set any engine you have installed as the default engine by right mouse clicking an engine, and then setting as default.

---

>> Just a note. The download and install UE5.3.2 downloaded and failed to install... I use Slackware, I am not surprised it failed.

This shouldn't happen because of Slackware, the install is just an zip file and the unzip code is built into the app.   So unless there was a network gltich causing the download to abort, it should have downloaded and installed it correctly.   I just installed 5.3.2 he other day and it worked correctly.  

---

>> Oh and for the record, Plugins have to be built in with the engine manually, although the asset manager did recognise them

It should be able to build them with the installing into an engine using that specific engine.  However, if you have v5.3.2 as your default engine, when you go to install it into a project, I think it will compile it using whatever the default engines code base. 
-  I need to look at the logic, now that I think about it, it should use the projects engine version, not the default engine version unless the project doesn't have its engine installed...

---
>> I thought you had a fancy way of installing plugins with linux... 
It should be fairly smart, however I just tested it and "install to project" seems broken.  
- I'll take a look at it and get it fixed.   It should be rebuilding it for that engine, but I think the logic is messed up somewhere....   Thanks for the report.

**Comment by DragonSai** [comment]
Thank you for responding so precisely .

I tried again tonight to install a downloaded asset into an existing project. I actually did a reboot as well.
It didn't go as planned.

None the less I love the work you have done for us linux game developers.

I will keep looking and posting if I notice other things.

**Comment by NathanaelA** [comment]
I've started working on several of the issues you mentioned, and a couple others I noticed.   I do believe [haven't finished testing] that a failed download still shows up as a "Valid" vault item, which is definitely an bug.

**Comment by NathanaelA** [comment]
I've reworked most this code over the last several versions.   Please  open a new ticket if you find anything still broken.  v0.28 was released today which should make everything a lot cleaner including nicer dialogs for installation.
